x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/MlNodeShutdownIT.java

```diff
@@ -100,6 +101,97 @@ public void testJobsVacateShuttingDownNode() throws Exception {
+        long weekAgo = now - 604800000;
+        long twoWeeksAgo = weekAgo - 604800000;
+        indexDocs(logger, "data", numDocs1, twoWeeksAgo, weekAgo);
+
+        // Open 6 jobs.  Since there are 3 nodes in the cluster we should get 2 jobs per node.
+
+        setupJobAndDatafeed("shutdown-close-job-1", ByteSizeValue.ofMb(2));
+        setupJobAndDatafeed("shutdown-close-job-2", ByteSizeValue.ofMb(2));
+        setupJobAndDatafeed("shutdown-close-job-3", ByteSizeValue.ofMb(2));
+        setupJobAndDatafeed("shutdown-close-job-4", ByteSizeValue.ofMb(2));
+        setupJobAndDatafeed("shutdown-close-job-5", ByteSizeValue.ofMb(2));
+        setupJobAndDatafeed("shutdown-close-job-6", ByteSizeValue.ofMb(2));
+
+        // Call the shutdown API for one non-master node.
+        String nodeNameToShutdown = Arrays.stream(internalCluster().getNodeNames())
+            .filter(nodeName -> internalCluster().getMasterName().equals(nodeName) == false).findFirst().get();
+        SetOnce<String> nodeIdToShutdown = new SetOnce<>();
+        SetOnce<String> jobIdToClose = new SetOnce<>();
+
+        assertBusy(() -> {
+            GetJobsStatsAction.Response statsResponse =
+                client().execute(GetJobsStatsAction.INSTANCE, new GetJobsStatsAction.Request(Metadata.ALL)).actionGet();
+            QueryPage<GetJobsStatsAction.Response.JobStats> jobStats = statsResponse.getResponse();
+            assertThat(jobStats, notNullValue());
+            long numJobsOnNodeToShutdown = jobStats.results().stream()
+                .filter(stats -> stats.getNode() != null && nodeNameToShutdown.equals(stats.getNode().getName())).count();
+            long numJobsOnOtherNodes = jobStats.results().stream()
+                .filter(stats -> stats.getNode() != null && nodeNameToShutdown.equals(stats.getNode().getName()) == false).count();
+            assertThat(numJobsOnNodeToShutdown, is(2L));
+            assertThat(numJobsOnOtherNodes, is(4L));
+            nodeIdToShutdown.set(jobStats.results().stream()
+                .filter(stats -> stats.getNode() != null && nodeNameToShutdown.equals(stats.getNode().getName()))
+                .map(stats -> stats.getNode().getId()).findFirst().get());
+            jobIdToClose.set(jobStats.results().stream()
+                .filter(stats -> stats.getNode() != null && nodeNameToShutdown.equals(stats.getNode().getName()))
+                .map(GetJobsStatsAction.Response.JobStats::getJobId).findAny().get());
+        });
+
+        client().execute(PutShutdownNodeAction.INSTANCE,
+            new PutShutdownNodeAction.Request(nodeIdToShutdown.get(), randomFrom(SingleNodeShutdownMetadata.Type.values()), "just testing"))
+            .actionGet();
+
+        if (randomBoolean()) {
+            // This isn't waiting for something to happen - just adding timing variation
+            // to increase the chance of subtle race conditions occurring.
+            Thread.sleep(randomIntBetween(1, 10));
+        }
```

Comment:
This may be good, but it seems to me that CI slowness/fastness might cause race conditions to bubble up themselves :)